pyblish_nukestudio/plugins/collect_selection.py

```diff
@@ -1,15 +1,13 @@
 import pyblish.api
 
-import hiero
 
-
-class CollectSelection(pyblish.api.Contextplugin):
+class CollectSelection(pyblish.api.ContextPlugin):
     """Inject the active project into context"""
 
-    version = (0, 1, 0)
     order = pyblish.api.CollectorOrder - 0.1
 
     def process(self, context):
+        import hiero
 
-        if hasattr(hiero, 'selection'):
-            context.set_data('selection', value=hiero.selection)
+        if hasattr(hiero, "selection"):
+            context.data["selection"] = hiero.selection
```

Comment:
When you export anything from NukeStudio, you need to select it.

You may be right about that we might be encouraging this interactive publishing, though think that might up each developer to decide how to handle that. The developer could for example tag the selection for later non-interactive publishing.
From the users perspective they would be able to do same thing as they are used to.